sdk/python/kfp/client/client.py

```diff
@@ -694,9 +694,9 @@ def _override_caching_options(self, pipeline_obj: dict,
 
     def list_pipelines(
         self,
-        page_token: str = '',
-        page_size: int = 10,
-        sort_by: str = '',
+        page_token: Optional[str] = '',
+        page_size: Optional[int] = 10,
+        sort_by: Optional[str] = '',
```

Comment:
`typing.Optional[<type>]` == `Union[<type>, None]` [[ref](https://docs.python.org/3/library/typing.html#typing.Optional)], which is not the same as a parameter having a default value.

Since the default is not `None` and (as far as I know) we don't permit passing `None`, the original annotation is correct.

Reply:
TIL. Thanks! I wonder if there is a way to specify optional-ness but exclude None, because it is optional from a UX standpoint, but maybe that doesn't matter?